src/sql-parser/src/parser.rs

```diff
@@ -4074,6 +4081,55 @@ impl<'a> Parser<'a> {
             } else {
                 Ok(WithOptionValue::Ident(ident!("secret")))
             }
+        } else if self.parse_keyword(ON) {
+            if self.parse_keyword(COMMIT) {
+                Ok(WithOptionValue::Refresh(RefreshOptionValue::OnCommit))
+            } else {
+                // This is needed when a user supplies `on` as a value to a completely different
+                // option (not even a REFRESH option).
+                Ok(WithOptionValue::Ident(ident!("on")))
+            }
+        } else if self.parse_keyword(AT) {
+            if self.parse_keyword(CREATION) {
+                Ok(WithOptionValue::Refresh(RefreshOptionValue::AtCreation))
+            } else if let Some(expr) = self.maybe_parse(Parser::parse_expr) {
+                Ok(WithOptionValue::Refresh(RefreshOptionValue::At(
+                    RefreshAtOptionValue { time: expr },
+                )))
+            } else {
+                Ok(WithOptionValue::Ident(ident!("at")))
+            }
+        } else if self.parse_keyword(EVERY) {
+            match self.maybe_parse(Parser::parse_value) {
+                Some(Value::String(interval)) => {
+                    let aligned_to = if self.parse_keywords(&[ALIGNED, TO]) {
+                        Some(self.parse_expr()?)
+                    } else {
+                        None
+                    };
+                    Ok(WithOptionValue::Refresh(RefreshOptionValue::Every(
+                        RefreshEveryOptionValue {
+                            interval,
+                            aligned_to,
+                        },
+                    )))
+                }
+                Some(v @ Value::Interval(_)) => {
+                    parser_err!(
+                            self,
+                            self.peek_prev_pos(),
+                            format!("Invalid value for REFRESH EVERY: `{v}`. The value should be a string parseable as an interval, e.g., '1 day'. The INTERVAL keyword should NOT be present!")
+                        )
+                }
+                Some(v) => {
+                    parser_err!(
+                            self,
+                            self.peek_prev_pos(),
+                            format!("Invalid value for REFRESH EVERY: `{v}`. The value should be a string parseable as an interval, e.g., '1 day'.")
+                        )
+                }
+                None => Ok(WithOptionValue::Ident(ident!("every"))),
+            }
```

Comment:
This should not be in `parse_option_value`, since these special syntaxes are not in general valid option values. That's why you have the special cases around `at` and `on`. Instead, you should factor out a method like `fn parse_materialized_view_refresh_option_value`, and then in `parse_materialized_view_option` you should sniff out `MaterializedViewOptionName::Refresh` and call `parse_materialized_view_refresh_option_value` instead of `parse_optional_option_value`.

Look at `parse_kafka_broker` for a good example of how this works for an existing special cased option!

Reply:
Ok, fixed!